API_GUIDE.md

```diff
@@ -143,11 +146,18 @@ PyTorch/XLA behaves semantically like regular PyTorch and XLA tensors, implement
     ```
 
 
-3.  Loops with a different number of iterations between steps are subject to similar observations as tensor shapes. PyTorch/XLA automatically handles them, but they are seen as different execution graphs and require recompilations.
+1.  Loops with a different number of iterations between steps are subject to similar observations as tensor shapes. PyTorch/XLA automatically handles them, but they are seen as different execution graphs and require recompilations. Therefore, to get the best performance, same computations should be run on all XLA devices in all hosts.
+One example of this is, iterators in `torch_xla_py.data_parallel` may drop the
+last few batches in the input iterator, in order to do the same amount of work
+on all XLA devices. In the extreme case where dataset is small, and there are
+too few steps, this may result in a no-op epoch.
 
-    `print(torch_xla._XLAC._xla_metrics_report())` can be used to print metrics at the end of each step to collect information regarding the number of compilations and operators that are part of the model but don’t have native XLA implementations. The `XLA_METRICS_FILE=1` environment setting can also be used to export per step metrics to a file.
+1. Sometimes model writers, when knowing that a PyTorch tensor is a scalar, they trigger `tensor.item()` (or equivalent PyTorch APIs which results to the same effects) calls, and they perform operations in Python scalar context, when similar operations can be performed using Pytorch tensor APIs. Following the latter approach will likely result in those operations behind fully fused within an XLA graph, without the need of issuing separate TPU computations.
+This can dramatically improve performance of the model, up to an N factor, where N is the number of `tensor.item()` calls per step.
 
-4. Sometimes model writers, when knowing that a PyTorch tensor is a scalar, they trigger `tensor.item()` (or equivalent PyTorch APIs which results to the same effects) calls, and they perform operations in Python scalar context, when similar operations can be performed using Pytorch tensor APIs. Following the latter approach will likely result in those operations behind fully fused within an XLA graph, without the need of issuing separate TPU computations.
+`print(torch_xla._XLAC._xla_metrics_report())` can be used to print metrics at the end of each step to collect information regarding the number of compilations and operators that are part of the model but don’t have native XLA implementations. The `XLA_METRICS_FILE=1` environment setting can also be used to export per step metrics to a file.
 
-   This can dramatically improve performance of the model, up to an N factor, where N is the number of `tensor.item()` calls per step.
+One common pitfall is, in this report, any counter that starts with `aten::`
```

Comment:
Is this really a pitfall? The counter merely indicates the pitfall, so I would drop the "One common pitfall is" and start the phrase with "In this report".